src/EtherFiNodesManager.sol

```diff
@@ -205,6 +205,72 @@ contract EtherFiNodesManager is
         node0.requestWithdrawal{value: msg.value}(pod, requests);
     }
 
+    /**
+     * @notice Triggers EIP-7251 consolidation requests, grouping by EigenPod automatically.
+     * @dev No `pod` param; we derive each pod from the validator pubkey using SSZ pubkey hash.
+     * @dev Access: only admin role, pausable, nonReentrant.
+     * @param requests Array of ConsolidationRequest:
+     *        - srcPubkey: 48-byte BLS pubkey of source validator
+     *        - targetPubkey: 48-byte BLS pubkey of target validator
+     *        - If srcPubkey == targetPubkey, this switches validator from 0x01 to 0x02 credentials
+     * @custom:fee Send enough ETH to cover sum of (feePerPod * requestsForPod). Overpay is OK.
+     */
+    function requestConsolidation(
+        IEigenPod.ConsolidationRequest[] calldata requests
+    ) external payable whenNotPaused nonReentrant onlyAdmin
+    {
+        // ------------ checks ------------
+        uint256 n = requests.length;
+        if (n == 0) revert EmptyConsolidationRequest();
+
+        bytes32 pkHash0 = calculateValidatorPubkeyHash(requests[0].srcPubkey);
```

Comment:
I think this is way more complicated than it needs to be, I believe you can drop all of the code ensuring everything is tied to the same pod, eigenlayer's contracts already verify this

```
            // Ensure target has verified withdrawal credentials pointed at this pod
            bytes32 sourcePubkeyHash = _calcPubkeyHash(request.srcPubkey);
            bytes32 targetPubkeyHash = _calcPubkeyHash(request.targetPubkey);
            ValidatorInfo memory target = validatorPubkeyHashToInfo(targetPubkeyHash);
            require(target.status == VALIDATOR_STATUS.ACTIVE, ValidatorNotActiveInPod());
```

Reply:
I believe they do not verify this but rather leave it to beacon chain to handle all the ownership related changes. EigenLayer trusts that if you submit an invalid request (validator doesn't belong to your pod), the beacon chain will simply _ignore/reject_ it.